src/shared/Validation.js

```diff
@@ -25,9 +25,80 @@ export const validateOnInput = (
   const formatErrors = sectionFormatErrors.slice();
 
   switch (fieldType) {
-    case 'number': {
+    case 'int16': {
       const idx = formatErrors.indexOf(FORM_ERRORS.INVALID_FORMAT);
-      if (input && !validator.isInt(input)) {
+      const validInt = validator.isInt(input);
+
+      let validIntType = true;
+      if (parseInt(input, 10) < -32768 || parseInt(input, 10) > 32767) {
```

Comment:
Avoid magic numbers. Create constants for these...

```
const INT_SHORT_SIGNED_MAX_VALUE = 32767;
const INT_SHORT_SIGNED_MIN_VALUE = -32768;
```

Reply:
Fixed